src/Nethermind.Int256.Benchmark/Benchmarks.cs

```diff
@@ -0,0 +1,505 @@
+using System.Collections.Generic;
+using System.Numerics;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Jobs;
+using Nethermind.Int256.Test;
+
+namespace Nethermind.Int256.Benchmark
+{
+    public class UnsingedBenchmarkBase
+    {
+        public IEnumerable<BigInteger> Values => System.Linq.Enumerable.Concat(new[] { Numbers.UInt256Max }, UnaryOps.RandomUnsinged(2));
+
+        public IEnumerable<UInt256> ValuesUint256 => Values.Select(x => (UInt256)x);
+
+        public IEnumerable<int> ValuesInt => UnaryOps.RandomInt(3);
+
+        public IEnumerable<UInt256> ValuesIntUint256 => ValuesInt.Select(x => (UInt256)x);
+    }
+
+    public class UnsignedIntTwoParamBenchmarkBase : UnsingedBenchmarkBase
+    {
+        [ParamsSource(nameof(Values))]
+        public BigInteger A;
+
+        [ParamsSource(nameof(ValuesUint256))]
+        public UInt256 UInt256A;
+
+        [ParamsSource(nameof(ValuesInt))]
+        public int D;
+
+        [ParamsSource(nameof(ValuesIntUint256))]
+        public UInt256 UInt256D;
+    }
+
+    public class UnsignedTwoParamBenchmarkBase : UnsingedBenchmarkBase
+    {
+        [ParamsSource(nameof(Values))]
+        public BigInteger A;
+
+        [ParamsSource(nameof(Values))]
+        public BigInteger B;
+
+        [ParamsSource(nameof(ValuesUint256))]
+        public UInt256 UInt256A;
+
+        [ParamsSource(nameof(ValuesUint256))]
+        public UInt256 UInt256B;
+    }
+
+    public class UnsignedThreeParamBenchmarkBase : UnsignedTwoParamBenchmarkBase
+    {
+        [ParamsSource(nameof(Values))]
+        public BigInteger C;
+
+        [ParamsSource(nameof(ValuesUint256))]
+        public UInt256 UInt256C;
+    }
+
+    public class SignedBenchmarkBase
+    {
+        public IEnumerable<BigInteger> Values => System.Linq.Enumerable.Concat(new[] { Numbers.UInt256Max }, UnaryOps.RandomUnsinged(2));
+
+        public IEnumerable<Int256> ValuesInt256 => Values.Select(x => (Int256)x);
+
+        public IEnumerable<int> ValuesInt => UnaryOps.RandomInt(3);
+
+        public IEnumerable<Int256> ValuesIntInt256 => ValuesInt.Select(x => (Int256)x);
+    }
+
+    public class SignedTwoParamBenchmarkBase : SignedBenchmarkBase
+    {
+        [ParamsSource(nameof(Values))]
```

Comment:
instead of 9 parameter combinations we end up with 81 parameter combinations which slows the benchmarks 9 times